Add 1 Tb disks for 6 worker node deploys for  /var/hadoop/data & /var/hadoop/temp

stvoutsin opened the pull request:
Add 1 Tb disks for 6 worker node deploys for  /var/hadoop/data & /var/hadoop/temp one for each.
Also add spark arrow enable param, add with both forms for now:
```
            spark.sql.execution.arrow.pyspark.enabled            true
            spark.sql.execution.arrow.enabled            true
```

Zarquan reviewed (commented):
A mixture of different things.

The target in the notes says 'Test deployment with pip libraries changes', but that is not what this PR is doing.

The PR contains settings for apache arrow mixed in with changes to Cinder discs for Hadoop.

If we are adding `spark.sql.execution.arrow.enabled`, then we should also include a test that checks for it.

Nigel has since closed the issue that requested `spark.sql.execution.arrow.enabled` (#824), so this is no longer required.

The PR is adding a third Cinder volume to the 6 worker deployments, but not to the 12 worker deployment? I'm not sure why we need to do this at this stage.

So far the only job that fails with Hadoop disc space issues is the large ingest using the 12 worker deployment (#817). If this is to see if we can complete the ingest by adding the extra disc space, then why not use the 12 worker deployment?

The deployment notes don't have anything to check that the extra discs end up being mounted in the right place. At a minimum we should run a `df` check on each directory on each of the workers.

The benchmark in the notes is the quick test that doesn't need extra space in `/var/hadoop/temp` or `/var/hadoop/data`. The benchmark results themselves are inconclusive, 9% slower for the mean proper motions, 18% faster for the source counts.

We may indeed end up needing these large Cinder volumes on our 6 worker deployments,
but we need to be clear why we are adding them and what if any cost in performance this has.

stvoutsin commented:
I've corrected the copy/paste typo on the target of the tests, and removed the extra spark arrow parameter.

Regarding moving the temp data on Cinder, I think its probably too late now to get conclusive tests on performance. 
My thoughts were that this should go through before next week, because we did see the current system run out of disk space due to temporary reduce data spilling onto /var/hadoop/temp with some of @NigelHambly experiments. While these experiments that triggered the issue were ingests, it still showed there are certain map/reduce jobs that will end up writing temporary data to that directory. If we don't include this in the deploy for next week, we are risking that any external users who try to use this may run a type of query that will cause our system to run out of disk space, and thus become unresponsive.  I think the possible performance hit is less important than the stability of the system, which is why I'd be leaning towards moving that directory to a larger disk. 

However as we need the live system setup today, I think we'll have to leave this PR & discussion & tests for sometime after next week. For now we'll have to go with what we have without these changes.

Zarquan commented on notes/stv/20220708-01-1Tb-disk-changes-test.txt:
```diff
@@ -0,0 +1,788 @@
+	},
+	"messages": [],
+	"time": {
+		"result": "SLOW",
+		"elapsed": "156.23",
+		"expected": "55.00",
+		"percent": "184.06",
```
This is what we need top look at.
Is this because we are using Cinder volumes for `/var/hadoop/temp`, or is this caused by something else.

Zarquan commented on notes/stv/20220708-01-1Tb-disk-changes-test.txt:
```diff
@@ -0,0 +1,788 @@
+	},
+	"messages": [],
+	"time": {
+		"result": "SLOW",
+		"elapsed": "52.05",
+		"expected": "22.00",
+		"percent": "136.57",
```
Something we need to look at before we merge this into the main deployments.

Zarquan commented:
@stvoutsin Do we need this ? If not, can we close it.
